plugins/store/src/store.rs

```diff
@@ -303,14 +337,152 @@ impl<R: Runtime> Store<R> {
+        let deleted = self.store.lock().unwrap().delete(key);
+        if deleted {
+            let _ = self.trigger_auto_save();
+        }
+        deleted
+    }
+
+    pub fn clear(&self) {
+        self.store.lock().unwrap().clear();
+        let _ = self.trigger_auto_save();
+    }
+
+    pub fn reset(&self) {
+        self.store.lock().unwrap().reset(&self.defaults);
+        let _ = self.trigger_auto_save();
+    }
+
+    pub fn keys(&self) -> Vec<String> {
+        self.store.lock().unwrap().keys().cloned().collect()
+    }
+
+    pub fn values(&self) -> Vec<JsonValue> {
+        self.store.lock().unwrap().values().cloned().collect()
+    }
+
+    pub fn entries(&self) -> Vec<(String, JsonValue)> {
+        self.store
+            .lock()
+            .unwrap()
+            .entries()
+            .map(|(k, v)| (k.to_owned(), v.to_owned()))
+            .collect()
+    }
+
+    pub fn length(&self) -> usize {
+        self.store.lock().unwrap().len()
+    }
+
+    pub fn load(&self) -> crate::Result<()> {
+        self.store.lock().unwrap().load(self.deserialize)
+    }
+
+    pub fn save(&self) -> crate::Result<()> {
+        self.store.lock().unwrap().save(self.serialize)
+    }
+
+    fn trigger_auto_save(&self) -> crate::Result<()> {
+        let Some(auto_save_delay) = self.auto_save else {
+            return Ok(());
+        };
+        if auto_save_delay.is_zero() {
+            return self.save();
+        }
+        let mut auto_save_debounce_sender = self.auto_save_debounce_sender.lock().unwrap();
+        if let Some(ref sender) = *auto_save_debounce_sender {
+            let _ = sender.send(AutoSaveMessage::Reset);
+            return Ok(());
+        }
+        let (sender, mut receiver) = unbounded_channel();
+        auto_save_debounce_sender.replace(sender);
+        drop(auto_save_debounce_sender);
+        let store = self.store.clone();
+        let serialize_fn = self.serialize;
+        let auto_save_debounce_sender = self.auto_save_debounce_sender.clone();
+        tauri::async_runtime::spawn(async move {
+            loop {
+                select! {
+                    should_cancel = receiver.recv() => {
+                        if matches!(should_cancel, Some(AutoSaveMessage::Cancel) | None) {
+                            return;
+                        }
+                    }
+                    _ = sleep(auto_save_delay) => {
+                        let _ = store.lock().unwrap().save(serialize_fn);
+                        auto_save_debounce_sender.lock().unwrap().take();
+                        return;
+                    }
+                };
+            }
+        });
+        Ok(())
+    }
+}
+
+impl<R: Runtime> Drop for Store<R> {
+    fn drop(&mut self) {
+        let auto_save_debounce_sender = self.auto_save_debounce_sender.lock().unwrap();
+        if let Some(ref sender) = *auto_save_debounce_sender {
+            let _ = sender.send(AutoSaveMessage::Cancel);
```

Comment:
are you sure we want to cancel on drop? shouldn't we force save here instead?

Reply:
I forget why I did this too be honest, we probably should move this to save instead, not sure we really need to do anything on drop